### PR TITLE
Renames based on Feedback from Service Team 

### PR DESCRIPTION
**Pull request by conniey**
Makes changes to EventSender and EventReceiver based on feedback from the Event Hubs service team.
* Rename EventSender to `EventHubProducer`, `EventReceiver` to `EventHubConsumer`, and their associated options.
* Rename "epoch"/"exclusiveReceiverPriority" to `ownerLevel`
* Add "consumerGroup" as a mandatory parameter when creating an EventHubConsumer.

### CODE REVIEW

File: core/azure-core/src/main/java/com/azure/core/amqp/AmqpSession.java [hunk 1]
```diff
@@ -27,26 +27,26 @@
     Duration getOpenTimeout();
 
     /**
-     * Creates a new AMQP sender link.
+     * Creates a new AMQP link that produces events and pushes them to the message broker.
```
Comment (jsquire) [inline core/azure-core/src/main/java/com/azure/core/amqp/AmqpSession.java]: nit:  The link publishes events, it doesn't produce/create them.... right?

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java [hunk 2]
```diff
@@ -14,85 +14,74 @@
...
     }
 
+
     /**
-     * Sets the exclusiveReceiverPriority value on this receiver. When populated, the priority indicates that a receiver is intended to be
-     * the only reader of events for the requested partition and an associated consumer group.
-     * To do so, this receiver will attempt to assert ownership over the partition; in the case where more than one
-     * exclusive receiver attempts to assert ownership for the same partition/consumer group pair, the one having a
-     * larger {@link EventReceiverOptions#exclusiveReceiverPriority()} value will "win".
+     * Sets the {@code ownerLevel} value on this consumer. When populated, the priority indicates that a consumer
```
Comment (jsquire) [inline eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java]: nit:  `the priority` should probably read `the level`   or `the level indicates the priority of...`

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java [hunk 3]
```diff
@@ -193,55 +160,37 @@ public Retry retry() {
     }
 
     /**
-     * Gets the priority for this receiver. If {@link Optional#isPresent()} is {@code false}, then this is not an
-     * exclusive receiver. Otherwise, it is and there can only be one receiver per (partition + consumer group)
-     * combination.
+     * Gets the priority for this consumer. If {@link Optional#isPresent()} is {@code false}, then this is not an
```
Comment (jsquire) [inline eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java]: nit:  Probably want to massage the `priority` nomenclature.   The level is an implied priority, but this reads pre-rename to me.

File: eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventHubConsumerOptionsTest.java [hunk 4]
```diff
@@ -89,19 +87,19 @@ public void invalidReceiverPriority() {
         // Arrange
         final long priority = 14;
```
Comment (jsquire) [inline eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventHubConsumerOptionsTest.java]: nit:  You may want to tweak the `priority` names here.

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java [hunk 5]
```diff
@@ -149,41 +155,78 @@ public EventSender createSender(EventSenderOptions options) {
...
         Objects.requireNonNull(eventPosition);
         Objects.requireNonNull(options);
 
-        final EventReceiverOptions clonedOptions = options.clone();
+        if (ImplUtils.isNullOrEmpty(consumerGroup)) {
+            throw new IllegalArgumentException("'partitionId' cannot be null or empty.");
+        }
+        if (ImplUtils.isNullOrEmpty(consumerGroup)) {
+            throw new IllegalArgumentException("'partitionId' cannot be null or empty.");
+        }
```
Comment (ramya-rao-a) [inline eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java]: Copy/paste issues here? we are checking for ConsumerGroup in both `if` blocks and throwing error for partitionId

Comment (conniey) [reply eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClient.java]: Ahaha phew. Good catch. I definitely copy and pasted that. 😆

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java [hunk 6]
```diff
@@ -193,55 +160,37 @@ public Retry retry() {
...
-     * combination.
+     * Gets the owner level for this consumer. If {@link Optional#isPresent()} is {@code false}, then this is not an
+     * exclusive consumer. Otherwise, it is an exclusive consumer, and there can only be one active consumer for each
+     * partition and consumer group combination.
      *
-     * @return An optional priority for this receiver.
+     * @return An optional owner level for this consumer.
      */
-    public Optional<Long> exclusiveReceiverPriority() {
+    public Optional<Long> ownerLevel() {
```
Comment (JonathanGiles) [inline eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java]: I'm not convinced `Optional` will add value here.

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java [hunk 7]
```diff
@@ -55,8 +55,20 @@ public EventHubClientBuilder() {
     /**
      * Sets the credential information given a connection string to the Event Hub instance.
      *
-     * @param connectionString The connection string to the Event Hub this client wishes to connect to. It is expected
-     *         that the Event Hub path and the shared key properties are contained in this connection string.
+     * <p>
+     * If the connection string is copied from the Event Hubs namespace, it will likely not contain the path to the
+     * desired Event Hub, which is needed. In this case, the path can be added manually by adding
+     * {@literal "EntityPath=[[ EVENT HUB NAME ]]"} to the end of the connection string. For example,
+     * ";EntityPath=telemetry-hub".
```
[the remark this answers is not in the document]
Comment (conniey) [reply eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubClientBuilder.java]: Lemme fix that. I copied it from @jsquire . Lol :D

File: eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java (hunk 6 above)
Comment (conniey) [reply eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/EventHubConsumerOptions.java]: Removed. :D

**Pull request merged**